updater/bootstrap.js

```diff
@@ -290,17 +290,51 @@ async function entry(info) {
         updaterArgs.push('-p');
         updaterArgs.push(statusWindow.webContents.getOSProcessId().toString());
     }
+    
+    const updaterStartCommand = `start \"\"  \"${updaterPath}\" `  
+    
+    updaterArgs.push('&& echo update_started_fine_app_can_be_closed');  
 
     log.info(updaterArgs);
 
-    cp.spawn(`${updaterPath}`, updaterArgs, {
+    const update_spawned = cp.spawn(`${updaterStartCommand}`, updaterArgs, {
         cwd: info.tempDir,
-        detached: true,
-        stdio: 'ignore',
+        detached: false,
+        //stdio: 'ignore',
         shell: true
     });
+    
+    log.info('updater process ' + `pid ${update_spawned.pid}`);  
 
-    return true;
+    //make promises for app exit , error , data and some timeout
+    const primiseExit = new Promise(resolve => {
+        update_spawned.on('exit', resolve);
+    });
+
+    const primiseError = new Promise(resolve => {
+        update_spawned.on('error', resolve);
+    });
+    
+    const primiseDataError = new Promise(resolve => {
+        update_spawned.stderr.on('data', resolve);
+    });
+
+    const primiseDataOut = new Promise(resolve => {
+        update_spawned.stdout.on('data', resolve);
+    });
+
+    //wait for something to happen
+    const promise = await Promise.race([primiseError, primiseExit, primiseDataOut, primiseDataError]); 
+    log.info('Updater spawn promise ' + `result \"${promise}\"`);
+    
+    update_spawned.unref();
+
+    if(`${promise}`.startsWith("update_started_fine_app_can_be_closed") )  //hardcoded string "update_started_fine_app_can_be_closed" 
```

Comment:
So I get what you're doing here, but this seems a little hacky and I want to make sure it's definitely needed.

My understanding is that you need this because you need to know whether the updater command had a `0` exit code, but if you listen to the `exit` signal on the child process, you're actually listening for the shell exit instead?  Will the shell always exit with `0` even if the command you ran failed?  Why do we need to spawn this in a shell?

Reply:
Yes. I checked on test app and it looks that we always get 0 if updater started. `&&` check exactly for 0 to start 'echo' or not. So no need to echo and check for a string.